django_json_ld/templatetags/json_ld.py

```diff
@@ -2,14 +2,37 @@
 
 from django import template
 from django.utils.safestring import mark_safe
+from django.template import TemplateSyntaxError
 
-from ..util import LazyEncoder
+from ..util import LazyEncoder, validate_sd, build_absolute_uri
+from .. import settings
 
 register = template.Library()
 
+empty_input_choice = settings.EMPTY_INPUT_RENDERING
 
-@register.simple_tag
-def render_json_ld(structured_data):
+
+@register.simple_tag(takes_context=True)
+def render_json_ld(context, structured_data):
+    # validate and throw error if sd is invalid
+    is_valid, err = validate_sd(structured_data)
+    if not is_valid:
+        raise TemplateSyntaxError(err)
+    # do empty input rendering stuff here
+    if len(structured_data) == 0:
+        if empty_input_choice == 'strict':
+            err = 'Empty structured_data provided under strict render policy.'
+            err += 'Consider change JSON_LD_EMPTY_INPUT_RENDERING setting to other values.'
```

Comment:
```suggestion
            err += ' Consider changing JSON_LD_EMPTY_INPUT_RENDERING setting to other values.'
```

Added a space so "Consider" is not glued to the full stop.